src/util.h

```diff
@@ -117,19 +117,14 @@ namespace rhost {
             }
         };
 
-        struct blob {
-            void* raw_data;
-            size_t length;
+        typedef std::vector<byte> blob_slice;
+        typedef std::vector<blob_slice> blob;
 
-            blob() :
-                raw_data(nullptr), length(0) {}
-            blob(void* const data, const size_t size) :
-                raw_data(data), length(size) {}
-
-            inline bool is_empty() { return (!raw_data) || (!length); }
-        };
-
-        typedef std::vector<blob> blobs;
+        inline void append_from_file(blob& blob, const std::string& path) {
+            std::ifstream ifs(path, std::ios::binary | std::ios::in);
+            std::noskipws(ifs);
+            blob.push_back(blob_slice(std::istream_iterator<byte>(ifs), std::istream_iterator<byte>()));
```

Comment:
This will actually read it byte by byte, so it'll be super slow (~2 orders of magnitude).

I think it's best to ditch iostreams here - they just aren't a good fit for binary data. Might as well just `fopen` and read the whole thing as a single chunk. `SCOPE_WARDEN` can be used to automatically clean up.